app/assets/javascripts/components/articles/article_graphs.jsx

```diff
@@ -1,170 +1,148 @@
-              graphWidth = {graphWidth}
-              graphHeight = {graphHeight}
-              articleData = {this.state.articleData}
-            />
-          );
-        } else {
-          graph = (
-            <EditSizeGraph
-              graphid ={this.graphId()}
-              graphWidth = {graphWidth}
-              graphHeight = {graphHeight}
-              articleData = {this.state.articleData}
-            />
-          );
-        }
-      } else {
-        editSize = (
-          <p>{I18n.t('articles.edit_size')}</p>
-        );
-        graph = (
-          <EditSizeGraph
-            graphid ={this.graphId()}
-            graphWidth = {graphWidth}
-            graphHeight = {graphHeight}
-            articleData = {this.state.articleData}
-          />
-        );
-      } // Display the loading element if articleData is not available
-     } else {
-      graph = <Loading />;
-    }
+  function handleHideGraph() {
+    setArticleData(null);
+    setShowGraph(false);
+  }
+
+  function handleRadioChange(event) {
+    setSelectedRadio(event.currentTarget.value);
+  }
 
-    return (
-      <a onClick={this.showGraph} className="inline">
-        {I18n.t('articles.article_development')}
-        <div className={className} ref={this.ref}>
-          <div className="radio-row">
-            {radioInput}
-            {editSize}
-          </div>
-          {graph}
+   const graphId = `vega-graph-${article_id}`;
+
+  const dataIncludesWp10 = articleData?.[0]?.wp10;
+
+  let graph;
+  if (!articleData) {
+    graph = <Loading />;
+  } else if (dataIncludesWp10 && selectedRadio === 'wp10_score') {
+    graph = (
+      <Wp10Graph
+        graphid={graphId}
+        graphWidth={500}
+        graphHeight={300}
+        articleData={articleData}
+      />
+    );
+  } else {
+    graph = (
+      <EditSizeGraph
+        graphid={graphId}
+        graphWidth={500}
+        graphHeight={300}
+        articleData={articleData}
+      />
+    );
+  }
+
+  let radioInput;
+
+  if (!dataIncludesWp10 || !articleData) {
+    radioInput = null;
+  } else {
+    radioInput = (
+      <div>
+        <div className="input-row">
+          <input
+            type="radio"
+            name="wp10_score"
+            id="wp10_score"
+            value="wp10_score"
+            checked={selectedRadio === 'wp10_score'}
+            onChange={handleRadioChange}
+          />
+          <label htmlFor="wp10_score">{I18n.t('articles.wp10')}</label>
+        </div>
+        <div className="input-row">
+          <input
+            type="radio"
+            name="edit_size"
+            id="edit_size"
+            value="edit_size"
+            checked={selectedRadio === 'edit_size'}
+            onChange={handleRadioChange}
+          />
+          <label htmlFor="edit_size">{I18n.t('articles.edit_size')}</label>
         </div>
-      </a>
+      </div>
     );
   }
-});
+
+  let editSize;
+  if (dataIncludesWp10 || !articleData) {
+    editSize = null;
```

Comment:
this is not required

Reply:
same here — this hasn’t been fixed and yet the conversation is marked as resolved.

Reply:
unresolved and fixed @TheTrio